File: src/main/java/com/dennisturco/dto/QuotationRequestDTO.java
package com.dennisturco.dto;

import java.time.LocalDate;

import jakarta.validation.constraints.Max;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.PositiveOrZero;
import jakarta.validation.constraints.Size;
import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class QuotationRequestDTO {

    private Long quotationStatusId;

    private Long customerId;

    @NotBlank(message = "Il numero del preventivo è obbligatorio")
    private String number;

    @NotNull
    @PositiveOrZero(message = "Non è possibile inserire un importo negativo")
    private Float amount;

    @NotNull
    @Min(value = 0, message = "IVA non può essere negativa")
    @Max(value = 100, message = "IVA non può essere > 100")
    private Integer vatPercentage;

    @NotNull
    @Min(value = 0, message = "Lo sconto non può essere negativo")
    @Max(value = 100, message = "Lo sconto non può essere > 100")
    private Integer discountPercentage;

    @Size(max = 2000, message = "Descrizione troppo lunga")
    private String description;

    @Size(max = 1000, message = "Note troppo lunghe")
    private String notes;

    private LocalDate issueDate;
    private LocalDate validityDate;
}

File: GestioPro.Common/DTOs/QuotationRequestDTO.cs
using System.ComponentModel.DataAnnotations;

namespace GestioPro.Common.DTOs;

public record QuotationRequestDTO(
    long CustomerId,
    long QuotationStatusId,
    [Required] string Number,
    [Required, Range(0, float.MaxValue)] float Amount,
    [Required, Range(0, 100)] int VatPercentage,
    [Required, Range(0, 100)] int DiscountPercentage,
    [MaxLength(2000)] string? Description,
    [MaxLength(1000)] string? Notes,
    DateOnly? IssueDate,
    DateOnly? ValidityDate
);
